mgmtsystem/azure.py

```diff
@@ -484,6 +593,36 @@ def current_ip_address(self, vm_name, resource_group=None):
         return etree.parse(StringIO(data)).getroot().xpath(
             "./Object/Property[@Name='IpAddress']/text()")
 
+    def list_free_nics(self, nic_template):
+        try:
+            azure_data = self.run_script(
+                """
+                Get-AzureRmNetworkInterface -ResourceGroupName {}|
```

Comment:
Again, indent all of these lines after """  run_script will dedent(script) so it doesn't have any affect.  Not that Powershell would even care.